spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java

```diff
@@ -67,7 +67,7 @@
 		JndiDataSourceAutoConfiguration.class,
 		XADataSourceAutoConfiguration.class })
 @AutoConfigureAfter(GcpContextAutoConfiguration.class)
-public abstract class GcpCloudSqlAutoConfiguration {
+public abstract class GcpCloudSqlAutoConfiguration { //NOSONAR must be abstract for HidePublicConstructorCheck
```

Comment:
That's not what the real problem is. 
We made it `abstract` for Checkstyle, but we can't honor SonarQube with that. The `//NOSONAR` is exactly for Sonar.
And we must explain in the comment why we do `//NOSONAR`.
There is indeed nothing to do with Checkstyle in this case.

Thanks for understanding.